ext/AvizExt/viz/coral_dynamics.jl

```diff
@@ -0,0 +1,275 @@
+    - `show_confints` : Show confidence intervals around series. Defaults to true.
+    - `colors` : Colormap for each taxonomy or scenario type. Defaults to Set1_5 for taxa and ADRIA defaults for scenario type.
+- `axis_opts` : Additional options to pass to adjust Axis attributes.
+  See: https://docs.makie.org/v0.19/api/index.html#Axis
+- `series_opts` : Additional options to pass to adjust Series attributes
+  See: https://docs.makie.org/v0.19/api/index.html#series!
+
+# Returns
+Figure or GridPosition
+"""
+function ADRIA.viz.taxonomy(
+    rs::ResultSet;
+    opts::Dict=Dict{Symbol,Any}(),
+    fig_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    axis_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    series_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}()
+)::Figure
+    if !haskey(rs.outcomes, :relative_taxa_cover)
+        throw(ArgumentError("Unable to found relative_taxa_cover in outcomes. This variable may be passed manually."))
+    end
+    return ADRIA.viz.taxonomy(
+        rs.inputs,
+        rs.outcomes[:relative_taxa_cover];
+        opts=opts,
+        fig_opts=fig_opts,
+        axis_opts=axis_opts,
+        series_opts=series_opts
+    )
+end
+function ADRIA.viz.taxonomy(
+    scenarios::DataFrame,
+    relative_taxa_cover::YAXArray;
+    opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    fig_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    axis_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    series_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}()
+)::Figure
+    fig_opts[:size] = get(fig_opts, :size, (1200, 1200))
+    f = Figure(; fig_opts...)
+
+    g = f[1, 1] = GridLayout()
+
+    _scenarios = copy(scenarios[1:end .∈ [relative_taxa_cover.scenarios], :])
+    scen_groups = if get(opts, :by_RCP, false)
+        ADRIA.analysis.scenario_rcps(_scenarios)
+    else
+        ADRIA.analysis.scenario_types(_scenarios)
+    end
+
+    ADRIA.viz.taxonomy!(
+        g,
+        relative_taxa_cover,
+        scen_groups,
+        opts=opts,
+        axis_opts=axis_opts,
+        series_opts=series_opts
+    )
+
+    return f
+end
+function ADRIA.viz.taxonomy!(
+    g::Union{GridLayout,GridPosition},
+    relative_taxa_cover::YAXArray,
+    scen_groups::Dict{Symbol, BitVector};
+    opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    axis_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    series_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}()
+)::Union{GridLayout,GridPosition}
+    axis_opts[:xlabel] = get(axis_opts, :xlabel, "Year")
+    axis_opts[:ylabel] = get(axis_opts, :ylabel, "Relative Cover")
+
+    show_confints::Bool = get(opts, :show_confints, true)
+    compare_taxa::Bool = get(opts, :compare_taxa, true)
+    if compare_taxa
+        # Create colors
+        n_groups::Int64 = length(relative_taxa_cover.taxa)
+        default_color = Symbol("Set1_" * string(n_groups))
+        color = get(opts, :colors, default_color)
+        _colors = categorical_colors(color, n_groups)
+
+        # Plot results
+        taxonomy_series_confint!(
```

Comment:
I think this function name can be misleading. It sounds like it will plot only the confint. 
Maybe renamed it to `taxonomy_by_intervention` ?